src/EnergyPlus/EcoRoofManager.cc

```diff
@@ -907,35 +909,47 @@ namespace EcoRoofManager {
         }
 
         // NEXT Add Precipitation to surface soil moisture variable (if a schedule exists)
-        if (!state.dataGlobal->WarmupFlag) {
-            state.dataEcoRoofMgr->CurrentPrecipitation = 0.0; // first initialize to zero
-        }
-        state.dataEcoRoofMgr->CurrentPrecipitation = 0.0; // first initialize to zero
-        if (state.dataWaterData->RainFall.ModeID == DataWater::RainfallMode::RainSchedDesign) {
-            state.dataEcoRoofMgr->CurrentPrecipitation = state.dataWaterData->RainFall.CurrentAmount; //  units of m
-            Moisture += state.dataEcoRoofMgr->CurrentPrecipitation / state.dataEcoRoofMgr->TopDepth;  // x (m) evenly put into top layer
-            if (!state.dataGlobal->WarmupFlag) {
-                state.dataEcoRoofMgr->CumPrecip += state.dataEcoRoofMgr->CurrentPrecipitation;
-            }
+        Moisture += state.dataEcoRoofMgr->CurrentPrecipitation / state.dataEcoRoofMgr->TopDepth; // x (m) evenly put into top layer
+
+        if (state.dataEnvrn->RunPeriodEnvironment) {
+            state.dataEcoRoofMgr->CumPrecip += state.dataEcoRoofMgr->CurrentPrecipitation;
         }
 
         // NEXT Add Irrigation to surface soil moisture variable (if a schedule exists)
         state.dataEcoRoofMgr->CurrentIrrigation = 0.0; // first initialize to zero
         state.dataWaterData->Irrigation.ActualAmount = 0.0;
-        if (state.dataWaterData->Irrigation.ModeID == DataWater::RainfallMode::IrrSchedDesign) {
+        if (state.dataWaterData->Irrigation.ModeID == DataWater::IrrigationMode::IrrSchedDesign) {
             state.dataEcoRoofMgr->CurrentIrrigation = state.dataWaterData->Irrigation.ScheduledAmount; // units of m
-            state.dataWaterData->Irrigation.ActualAmount = state.dataEcoRoofMgr->CurrentIrrigation;
+            if (state.dataWaterData->UsePrecipitation) {
+                state.dataWaterData->Irrigation.ActualAmount = state.dataEcoRoofMgr->CurrentIrrigation + state.dataEcoRoofMgr->CurrentPrecipitation;
```

Comment:
This is double-counting `CurrentPrecipitation`. When I suggested adding the new field, I did not understand that Site:Precipitation was already being applied to the EcoRoof (but only through Site:Precipitation, not from the epw). As I look at this today:
1. I think we should remove the new field from RoofIrrigation.
2. If we think the users need to be able to turn off precipitation, then the new field belongs at the end of Material:RoofVegetation and it should default to Yes.
3. Sorry to change direction at the last minute.

Reply:
Thanks. Earlier I thought the flag just add the rainfall to the reported overall irrigation, as people might think rain adds water to the plant in addition to irrigation pipes. But maybe they won't interpret it this way. So I guess we should remove this new flag. Sorry for the misunderstanding. I didn't document it clearly.